modules/database/src/admin/schema.admin.ts

```diff
@@ -38,45 +38,57 @@ export class SchemaAdmin {
             updatedAt: 1,
           },
         },
-      )
-      .then(r => {
-        return { schemas: r };
-      });
+      );
+    const cndSchemas = await this.database
+      .getSchemaModel('_DeclaredSchema')
+      .model.findMany(
+        { 'modelOptions.conduit.cms': { $exists: false } },
+        {
+          select: 'name extensions',
+          sort: {
+            updatedAt: 1,
+          },
+        },
+      );
+    return { schemas: cmsSchemas.concat(cndSchemas) };
   }
 
-  async importCustomSchemas(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
+  async importSchemas(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
     const schemas = call.request.params.schemas;
     for (const schema of schemas) {
-      const existingSchema = await this.database
-        .getSchemaModel('_DeclaredSchema')
-        .model.findOne({ name: schema.name });
-      const operation = isNil(existingSchema) ? 'create' : 'update';
-      const imported = schema.modelOptions.conduit.imported === true;
-      const modelOptions = SchemaConverter.getModelOptions({
-        cmsSchema: true,
-        existingModelOptions: schema.modelOptions,
-        importedSchema: imported,
-      });
-      try {
-        validateSchemaInput(schema.name, schema.fields, modelOptions);
-      } catch (err: unknown) {
-        throw new GrpcError(status.INTERNAL, (err as Error).message);
+      const isCmsSchema = schema.modelOptions;
```

Comment:
No, you're right about `modelOptions` (righfully) not being exported.
I read that part wrong (confused with db schema entries.

This technically works as it is, but we should be crosschecking whether imported schemas are actually module schemas or not regardless of what's provided in the request body.

What if I were to provide `modelOptions` in a module schema?
We're currently treating schema names as unique, therefore we could compare with the ones available in the adapter. If unknown or `schema.modelOptions.cms`, treat as CMS schema, otherwise throw early with an appropriate error message.
These checks would ideally take place before any modifications have taken place.